backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchIndexer.java

```diff
@@ -103,26 +105,34 @@ public IndexResponse indexMessage(String id, String content) {
             .get();
     }
 
-    public BulkResponse updateMessages(List<UpdatedRepresentation> updatedDocumentParts) {
-        Preconditions.checkNotNull(updatedDocumentParts);
-        BulkRequestBuilder bulkRequestBuilder = client.prepareBulk();
-        updatedDocumentParts.forEach(updatedDocumentPart -> bulkRequestBuilder.add(
-            client.prepareUpdate(
-                indexName.getValue(),
-                typeName.getValue(),
-                updatedDocumentPart.getId())
-            .setDoc(updatedDocumentPart.getUpdatedDocumentPart())));
-        return bulkRequestBuilder.get();
+    public Optional<BulkResponse> updateMessages(List<UpdatedRepresentation> updatedDocumentParts) {
+        try {
+            Preconditions.checkNotNull(updatedDocumentParts);
+            BulkRequestBuilder bulkRequestBuilder = client.prepareBulk();
+            updatedDocumentParts.forEach(updatedDocumentPart -> bulkRequestBuilder.add(
+                client.prepareUpdate(
+                    indexName.getValue(),
+                    typeName.getValue(),
+                    updatedDocumentPart.getId())
+                .setDoc(updatedDocumentPart.getUpdatedDocumentPart())));
+            return Optional.of(bulkRequestBuilder.get());
+        } catch (ValidationException e) {
```

Comment:
No logging?
Do you know why the indexing can fail?
If the exception is issued during the forEach, it means other parts are ignored?